### PR TITLE
fix(middleware-ssec): ssecMiddleware with strict base64 validation

### PR DESCRIPTION
**Pull request by RanVaknin**
### Issue
#5863

### Description

Making a a more robust check for if the provided key is indeed a base64  by comparing the byte length to 32, since AES-256 specifies a cipher must be 256bits (32 bytes) we can assert that the cipher provided must have a byte length of 32.

> AES-256() The block cipher specifed in this Standard with 256-bit keys

https://csrc.nist.gov/pubs/fips/197/final

### Testing
Added unit test coverage to cover the more of the middleware changes from https://github.com/aws/aws-sdk-js-v3/pull/5676
And additionally unit tests to cover the `isValidBase64EncodedSSECustomerKey` helper function.

### Driver code:
Borrowed from edge case mentioned in #5863:

```js
async function putObjectWithMD5EncodedSSECKey(){
  try {
    const SSECustomerKey = crypto
    .createHash("md5")
    .update("foo")
    .digest("hex");
    const SSECustomerKeyMD5 = crypto
  .createHash("md5")
  .update(SSECustomerKey)
  .digest("hex");

    const response = await client.send(new PutObjectCommand({
      Bucket: 'testbucket',
       Body: 'This is a test',      
       Key: "foo",    
       SSECustomerKey: SSECustomerKey,
      SSECustomerAlgorithm: 'AES256',
      SSECustomerKeyMD5: SSECustomerKeyMD5
    }))
    console.log(response.$metadata.httpStatusCode);
  } catch (error) {
    console.error(error);
  }
}

// 200
```
### Additional context
Add any other context about the PR here.

### Checklist
- [ ] If you wrote E2E tests, are they resilient to concurrent I/O?
- [ ] If adding new public functions, did you add the `@public` tag and enable doc generation on the package?

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-ssec/src/index.ts [hunk 1]
```diff
@@ -78,3 +77,19 @@ export const getSsecPlugin = (config: PreviouslyResolved): Pluggable<any, any> =
...
   },
 });
+
+function isValidBase64(str: string) {
+  const base64Regex = /^(?:[A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/;
+
+  if (!base64Regex.test(str)) return false;
+
+  try {
+    const decodedBytes = Buffer.from(str, "base64");
```
Comment (kuhe) [inline packages/middleware-ssec/src/index.ts:87]: encoding/decoding must use the SerdeContext's encoder/decoders, i.e. `options` above. 

Buffer is not available in browsers.

File: packages/middleware-ssec/src/index.ts [hunk 2]
```diff
@@ -78,3 +77,19 @@ export const getSsecPlugin = (config: PreviouslyResolved): Pluggable<any, any> =
     clientStack.add(ssecMiddleware(config), ssecMiddlewareOptions);
   },
 });
+
+function isValidBase64(str: string) {
```
Comment (kuhe) [inline packages/middleware-ssec/src/index.ts:81]: This function should be named after its logical behavior. It is not checking base64 validity.

It is checking whether the input is a base64 encoded SSE customer key.

File: packages/middleware-ssec/src/index.ts [hunk 3]
```diff
@@ -78,3 +77,16 @@ export const getSsecPlugin = (config: PreviouslyResolved): Pluggable<any, any> =
     clientStack.add(ssecMiddleware(config), ssecMiddlewareOptions);
   },
 });
+
+function isValidBase64EncodedSSECustomerKey(str: string, options: PreviouslyResolved) {
```
Comment (kuhe) [inline packages/middleware-ssec/src/index.ts:81]: please add two unit or integration tests, one with a passing example and one with a failing example

**Review by kuhe — APPROVED** [review]
